zipkin-lens/src/components/DependenciesPage/DependenciesGraph.jsx

```diff
@@ -0,0 +1,180 @@
+  control: base => ({
+    ...base,
+    width: '15rem',
+  }),
+};
+
+const vizStyle = {
+  colorText: theme.palette.primary.contrastText,
+  colorTextDisabled: theme.palette.primary.contrastText,
+  colorConnectionLine: theme.palette.grey[800],
+  colorTraffic: {
+    normal: theme.palette.primary.dark,
+    warning: theme.palette.secondary.dark,
+    danger: theme.palette.secondary.dark,
+  },
+  colorDonutInternalColor: theme.palette.primary.light,
+  colorDonutInternalColorHighlighted: theme.palette.primary.light,
+  colorLabelBorder: theme.palette.primary.dark,
+  colorLabelText: theme.palette.primary.contrastText,
+  colorTrafficHighlighted: {
+    normal: theme.palette.primary.main,
+  },
+};
+
+const propTypes = {
+  selectedNodeName: PropTypes.string,
+  onNodeClick: PropTypes.func.isRequired,
+  edges: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+  nodes: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+};
+
+const defaultProps = {
+  selectedNodeName: '',
+};
+
+const DependenciesGraph = React.memo(({
+  selectedNodeName,
+  onNodeClick,
+  edges,
+  nodes,
+}) => {
+  const classes = useStyles();
+  const [filter, selectFilter] = useReducer((_, selected) => (selected ? selected.value : ''), '');
+
+  const filterConnections = useCallback((object, value) => {
+    if (!value) {
+      return true;
+    }
+    if (object.name === value) {
+      return true;
+    }
+    return object.source.name === value || object.target.name === value;
+  }, []);
+
+  const filterNodes = useCallback((object, value) => {
+    if (!value) {
+      return true;
+    }
+    if (object.name === value) {
+      return true;
+    }
+    return object.incomingConnections.find(conn => conn.source.name === value)
+      || object.outgoingConnections.find(conn => conn.target.name === value);
+  }, []);
+
+  const handleObjectHighlight = useCallback((highlightedObject) => {
+    if (!highlightedObject) {
+      onNodeClick(null);
+      return;
+    }
+    if (highlightedObject.type === 'node' && highlightedObject.getName() !== selectedNodeName) {
+      onNodeClick(highlightedObject.getName());
+    }
+  }, [onNodeClick, selectedNodeName]);
+
+  const maxVolume = useMemo(() => {
+    if (edges.length > 0) {
+      return edges.map(edge => edge.metrics.normal + edge.metrics.danger)
+        .reduce((a, b) => Math.max(a, b));
+    }
+    return 0;
+  }, [edges]);
+
+  const filterOptions = useMemo(() => nodes.map(node => ({
+    value: node.name,
+    label: node.name,
+  })), [nodes]);
+
+  return (
+    <Box className={classes.root}>
+      <VizceralExt
+        allowDraggingOfNodes
+        targetFramerate={30}
+        traffic={{
+          renderer: 'region',
+          layout: 'ltrTree',
+          name: 'dependencies-graph',
+          maxVolume: maxVolume * 50,
+          nodes,
+          connections: edges,
+        }}
+        objectHighlighted={handleObjectHighlight}
+        styles={vizStyle}
+        key={filter}
```

Comment:
Ah do we need this anymore? I'm hoping that using vizceral's `filter` feature fixes the issue we were seeing. If not, we can keep it.

Reply:
If we do need to keep it, we should still add the comment right here
```
// For some reason, when updating `filters`, vizceral does not rerender the graph correctly in place. So instead of rerendering in place, we replace the `key`, which causes react to completely destroy and reconstruct the component, effectively rerendering the graph from scratch.
```

Reply:
Yes, we need this yet...
Without this, the layout will not be reset...